core/src/main/java/org/apache/carbondata/core/metadata/schema/table/DataMapSchema.java

```diff
@@ -41,6 +48,24 @@
    */
   private Map<String, String> properties;
 
+  /**
+   * map of parent column name to set of child column column without
+   * aggregation function
+   */
+  private Map<String, Set<ColumnSchema>> parentToNonAggChildMapping;
+
+  /**
+   * map of parent column name to set of child columns column with
+   * aggregation function
+   */
+  private Map<String, Set<ColumnSchema>> parentToAggChildMapping;
+
+  /**
+   * map of parent column name to set of aggregation function applied in
+   * in parent column
+   */
+  private Map<String, Set<String>> parentColumnToAggregationsMapping;
+
   public DataMapSchema(String className) {
```

Comment:
Create a factory and extend this datamap schema and implement as per the class name. All aggdatamap related should go to AggregationDataMapSchema class. DataMapSchema should be the generic class and it should only contains the generic attributes.